### PR TITLE
Add keyless integration guide

### PR DESCRIPTION
**Pull request by heliuchuan**
### Checklist

- [x] Do all Lints pass?
- [x] Have you ran `pnpm spellcheck`?
- [x] Have you ran `pnpm fmt`?
- [x] Have you ran `pnpm lint`?

### CODE REVIEW

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+3. Select or create your OAuth 2.0 Client ID
+4. Configure authorized origin (your dApp origin)
+5. Configure redirect URIs (the handler for the callback after authentication which will receive the authorization code and/or id_token)
+6. Obtain the `client_id` of your application
+
+## Step 2. Install the Aptos TypeScript SDK
+
+```bash
+# Experimental SDK version with Keyless support.
+npm install @aptos-labs/ts-sdk@zeta
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:84]:
```suggestion
pnpm install @aptos-labs/ts-sdk@zeta
```

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+1. A user’s email address is not revealed on-chain to anybody, including other users and validators.
+2. A user’s blockchain address and associated transaction history is hidden from the identity provider (e.g. Google).
+
+Keyless accounts are revolutionary to users for the following reasons:
+
+1. “1-click” account creation via familiar Web2 logins like `Sign In with Google`.
+2. Ability to transact on the Aptos blockchain without needing to navigate away from the application experience to download a wallet.
+3. Requires no secret key management by the user. This means blockchain account access is synonymous with access to one’s OIDC account and Web2-like recovery flows are available to regain access to one’s blockchain account in case the user ever loses access to their OIDC account.
+4. Seamless cross-devices experiences; users log in with their OIDC account no matter what device they are on - no need to download wallet software on each device, import their keys and encrypt them with a password, which must be maintained.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:17]: Nit: cross-device

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
+---
+title: "Aptos Keyless Integration Guide"
+---
+
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:4]: Add a top Header like
```
# Aptos Keyless
```

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+2. Ability to transact on the Aptos blockchain without needing to navigate away from the application experience to download a wallet.
+3. Requires no secret key management by the user. This means blockchain account access is synonymous with access to one’s OIDC account and Web2-like recovery flows are available to regain access to one’s blockchain account in case the user ever loses access to their OIDC account.
+4. Seamless cross-devices experiences; users log in with their OIDC account no matter what device they are on - no need to download wallet software on each device, import their keys and encrypt them with a password, which must be maintained.
+
+:::tip Keyless Account Scoping
+Use of the **_Aptos Keyless Integration Guide_** will allow for the integration of keyless accounts directly into your application. This means that blockchain accounts created on your application are scoped to the domain of your application (logging in with your Google account on dApp A and logging in with your Google account on dApp B will create separate accounts). Stay tuned for more to come on Aptos’ plan to allow Keyless accounts to be used portably across applications.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:20]: Nit: 
This means that blockchain accounts created on your application are scoped to the domain of your application
This means that blockchain accounts are scoped to your application's domain

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+:::
+
+_Note: this guide is oriented toward non-wallet applications. If you are a wallet developer and have interest in using Keyless accounts, please reach out to us directly._
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:23]: Nit: capitalize This

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+### Supported Identity Providers
+
+Currently only Google is supported. We will support additional OIDC providers in the future (e.g., Apple, Kakaotalk, Microsoft, etc.).
+
+| Identity Provider | Auth URL                                                                                                                                                                       |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Google            | https://accounts.google.com/o/oauth2/v2/auth/oauthchooseaccount?client_id=${CLIENT_ID}&redirect_uri=${REDIRECT_URI}&response_type=id_token&scope=openid%20email&nonce=${NONCE} |
+
+### Google
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:67]: My context may be old, but if not mention somewhere that dapps need to use the implicit OAuth flow

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+        const clientId = env.IDP_CLIENT_ID
+        // Get the nonce associated with ephemeralKeyPair
+        const nonce = ephemeralKeyPair.nonce
+        ```
+
+    4. Construct the login URL for the user to authenticate with the IdP.  Make sure the `openid` scope is set.  Other scopes such as `email` and `profile` can be set based on your app’s needs.
+
+        ```jsx
+        const loginUrl = `https://accounts.google.com/o/oauth2/v2/auth?response_type=id_token&scope=openid+email+profile&nonce=${nonce}&redirect_uri=${redirectUri}&client_id=${clientId}`
+        ```
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:126]: make these all tsx

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+    1. In the background, we create an ephemeral key pair. Store this in local storage.
+
+        ```tsx
+        import {EphemeralKeyPair} from '@aptos-labs/ts-sdk';
+
+        const ephemeralKeyPair = EphemeralKeyPair.generate();
+        ```
+
+    2. Save the `EphemeralKeyPair` in local storage keyed by its `nonce`.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:106]: Nit: "storage, keyed"

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+        ```jsx
+        // This saves the EphemeralKeyPair in local storage keyed by the nonce.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:109]: Same as above

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+        storeEphemeralKeyPair(ephemeralKeyPair.nonce, ephemeralKeyPair);
+        ```
+
+    3. Prepare the URL params of the login URL.  Set the `redirect_uri` and `client_id` to your configured values with the IdP.  Set the `nonce` to the nonce of the `EphemeralKeyPair` from step 1.i.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:113]: Remove the extra space between `URL.  Set` and `IdP.  Set`

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+
+        ```jsx
+        const redirectUri = 'https://.../login/callback'
+        const clientId = env.IDP_CLIENT_ID
+        // Get the nonce associated with ephemeralKeyPair
+        const nonce = ephemeralKeyPair.nonce
+        ```
+
+    4. Construct the login URL for the user to authenticate with the IdP.  Make sure the `openid` scope is set.  Other scopes such as `email` and `profile` can be set based on your app’s needs.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:122]: Same as above

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+
+        ```jsx
+        const loginUrl = `https://accounts.google.com/o/oauth2/v2/auth?response_type=id_token&scope=openid+email+profile&nonce=${nonce}&redirect_uri=${redirectUri}&client_id=${clientId}`
+        ```
+
+    5. When the user clicks the login button, redirect the user to the `loginUrl` that was created in step 1.iv.
+
+### 2. Handle the callback by parsing the token and create a Keyless account for the user.
+
+    1. Once the user completes the login flow, they will be redirected to the `redirect_uri` set in step 1. The JWT will be set in the URL as a search parameter in a URL fragment keyed by `id_token`. Extract the JWT from the `window` by doing the following:
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:132]: Nit: `fragment, keyed`

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+
+        ```jsx
+        import {Aptos} from '@aptos-labs/ts-sdk';
+
+        const aptos = new Aptos();
+        const keylessAccount = await aptos.deriveKeylessAccount({
+            jwt,
+            ephemeralKeyPair,
+        });
+        ```
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:172]: `tsx` not `jsx`

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+
+    4. Construct the login URL for the user to authenticate with the IdP.  Make sure the `openid` scope is set.  Other scopes such as `email` and `profile` can be set based on your app’s needs.
+
+        ```jsx
+        const loginUrl = `https://accounts.google.com/o/oauth2/v2/auth?response_type=id_token&scope=openid+email+profile&nonce=${nonce}&redirect_uri=${redirectUri}&client_id=${clientId}`
+        ```
+
+    5. When the user clicks the login button, redirect the user to the `loginUrl` that was created in step 1.iv.
+
+### 2. Handle the callback by parsing the token and create a Keyless account for the user.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:130]: Nit: Remove period at end

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+        import {Aptos} from '@aptos-labs/ts-sdk';
+
+        const aptos = new Aptos();
+        const keylessAccount = await aptos.deriveKeylessAccount({
+            jwt,
+            ephemeralKeyPair,
+        });
+        ```
+
+### 3. Submit transactions to the Aptos blockchain.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:174]: Nit: Remove period at end

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
+    2. Sign and submit the transaction to the chain.
+
+        ```jsx
+        const committedTxn = await aptos.signAndSubmitTransaction({ signer: keylessAccount, transaction });
+        ```
+
+    3. Wait for the transaction to be processed on-chain
+
+        ```jsx
+        const committedTransactionResponse = await aptos.waitForTransaction({ transactionHash: committedTxn.hash });
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:198]: `tsx` not `jsx`

**Review by hariria — COMMENTED** [review]
Grammar and punctuation mostly. Maybe mention something about implicit flow if thats still required and include a link reference for those who are unfamilar

File: apps/docusaurus/docs/guides/keyless-accounts.md (hunk 11 above)
Comment (heliuchuan) [reply apps/docusaurus/docs/guides/keyless-accounts.md:216]: I'm a double space kinda guy haha

**Review by alinush — APPROVED** [review]
🙈

File: apps/docusaurus/docs/guides/keyless-accounts.md [hunk 17]
```diff
@@ -72,6 +72,8 @@ Currently only Google is supported. We will support additional OIDC providers in
 | ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
 | Google            | https://accounts.google.com/o/oauth2/v2/auth/oauthchooseaccount?client_id=${CLIENT_ID}&redirect_uri=${REDIRECT_URI}&response_type=id_token&scope=openid%20email&nonce=${NONCE} |
 
+The impilicit flow (no authorization code exchange) is the preferred method of authentication.  The integration steps assumes use of implicit flow.
```
Comment (hariria) [inline apps/docusaurus/docs/guides/keyless-accounts.md:75]: Nit: impilicit -> implicit
Nit: assumes -> assume

**Review by hariria — APPROVED** [review]

**Pull request merged**